packages/pluggableWidgets/tree-view-web/src/components/__tests__/TreeView.spec.tsx

```diff
@@ -182,4 +186,122 @@ describe("TreeView", () => {
         expect(getExpandIconFromBranchHeader(updatedFirstHeader)).toHaveLength(0);
         expect(getCollapseImageFromBranchHeader(updatedFirstHeader)).toHaveLength(1);
     });
+
+    it("is not clickable and doesn't show an icon when its child has no nodes", () => {
+        const nestedItems: TreeViewProps["items"] = [
+            {
+                id: "11" as GUID,
+                value: "Parent treeview with a nested treeview that is empty",
+                content: (
+                    <TreeView
+                        {...defaultProps}
+                        class=""
+                        items={[]}
+                        isUserDefinedLeafNode={false}
+                        startExpanded={false}
+                    />
+                )
+            }
+        ];
+
+        const treeView = mount(
+            <TreeView
+                {...defaultProps}
+                {...customIconProps}
+                class=""
+                items={nestedItems}
+                isUserDefinedLeafNode={false}
+                startExpanded
+            />
+        );
+
+        const parentTreeViewHeader = treeView.findWhere(
+            node => node.type() === "header" && node.text().includes("Parent treeview")
+        );
+        expect(parentTreeViewHeader).toHaveLength(1);
+        expect(parentTreeViewHeader.prop("role")).not.toBe("button");
```

Comment:
Can we check here whether the role is actually not on the DOM node itself instead of checking the prop?

Reply:
I will try to see what I can do

Reply:
Nice suggestion! I used `.getDOMNode().getAttribute()` now, which should be a better verification than just using `.prop`: 76fe4ab